sdk/src/jumbf_io.rs

```diff
@@ -408,45 +413,146 @@ pub mod tests {
+        let mut writer = Cursor::new(Vec::new());
+        let embed_ref = RemoteRefEmbedType::Xmp(REMOTE_URL.to_string());
+        remote_ref_writer
+            .embed_reference_to_stream(reader, &mut writer, embed_ref)
+            .unwrap();
+        writer.set_position(0);
+        let xmp = asset_handler.get_reader().read_xmp(&mut writer).unwrap();
+        let loaded = crate::utils::xmp_inmemory_utils::extract_provenance(&xmp).unwrap();
+        assert_eq!(loaded, REMOTE_URL.to_string());
+    }
+
+    #[test]
+    fn test_streams_jpeg() {
+        let mut reader = std::fs::File::open("tests/fixtures/IMG_0003.jpg").unwrap();
+        test_jumbf("jpeg", &mut reader);
+        reader.rewind().unwrap();
+        test_remote_ref("jpeg", &mut reader);
+    }
+
+    #[test]
+    fn test_streams_png() {
+        let mut reader = std::fs::File::open("tests/fixtures/sample1.png").unwrap();
+        test_jumbf("png", &mut reader);
+        reader.rewind().unwrap();
+        test_remote_ref("png", &mut reader);
+    }
+
+    #[test]
+    fn test_streams_webp() {
+        let mut reader = std::fs::File::open("tests/fixtures/sample1.webp").unwrap();
+        test_jumbf("webp", &mut reader);
+        reader.rewind().unwrap();
+        test_remote_ref("webp", &mut reader);
+    }
+
+    #[test]
+    fn test_streams_wav() {
+        let mut reader = std::fs::File::open("tests/fixtures/sample1.wav").unwrap();
+        test_jumbf("wav", &mut reader);
+        reader.rewind().unwrap();
+        test_remote_ref("wav", &mut reader);
+    }
+
+    #[test]
+    fn test_streams_avi() {
+        let mut reader = std::fs::File::open("tests/fixtures/test.avi").unwrap();
+        test_jumbf("avi", &mut reader);
+        //reader.rewind().unwrap();
+        //test_remote_ref("avi", &mut reader); // not working
+    }
+
+    #[test]
+    fn test_streams_tiff() {
+        let mut reader = std::fs::File::open("tests/fixtures/TUSCANY.TIF").unwrap();
+        test_jumbf("tiff", &mut reader);
+        reader.rewind().unwrap();
+        test_remote_ref("tiff", &mut reader);
+    }
+
+    #[test]
+    fn test_streams_svg() {
+        let mut reader = std::fs::File::open("tests/fixtures/sample1.svg").unwrap();
+        test_jumbf("svg", &mut reader);
+        //reader.rewind().unwrap();
+        //test_remote_ref("svg", &mut reader); // svg doesn't support remote refs
+    }
+
+    #[test]
+    fn test_streams_mp3() {
+        let mut reader = std::fs::File::open("tests/fixtures/sample1.mp3").unwrap();
+        test_jumbf("mp3", &mut reader);
+        // mp3 doesn't support remote refs
+        //reader.rewind().unwrap();
+        //test_remote_ref("mp3", &mut reader); // not working
+    }
+
+    #[test]
+    fn test_streams_avif() {
+        let mut reader = std::fs::File::open("tests/fixtures/sample1.avif").unwrap();
+        test_jumbf("avif", &mut reader);
+        //reader.rewind().unwrap();
+        //test_remote_ref("avif", &mut reader);  // not working
```

Comment:
One thing you can do is mark this test as _ignored_ - that way the unit test output will show that we ignored this test, instead of commenting out the unsupported assertion. Maybe separate the assertions _test_jumbf_ and _test_remote_ref_ into two test cases and ignore the unsupported part? 

```rust
#[test]
#[ignore]
fn test_streams_remote_ref() {
    let mut reader = std::fs::File::open("tests/fixtures/sample1.avif").unwrap();
    reader.rewind().unwrap();
    test_remote_ref("avif", &mut reader);  // not working
}
```

Reply:
The better thing would be to fix the code to support it as it should. I tried that and ran out of time. So I'll continute with this as is and get back to it.